clients/cobol-lsp-vscode-extension/src/services/CopybooksDownloader.ts

```diff
@@ -54,22 +56,25 @@ export class CopybooksDownloader implements vscode.Disposable {
             return;
         }
         const depDesc: DependenciesDesc = loadDepFile(depFileUri);
+
+        // TODO: REMOVE PROFILE OR ADD IT AFTER LOCAL CHECK
         const profile: string = await this.profileService.getProfile(depDesc.programName);
         if (!profile) {
             return;
         }
 
-        const missingCopybooks: string[] = await this.listMissingCopybooks(depDesc.copybooks, profile);
+        this.prioritizer.checkCopybooksPresentLocal(depDesc.copybooks,
+        vscode.workspace.getConfiguration(SETTINGS_SECTION).get(PATHS_LOCAL_KEY));
+        const notLocalCpy: string[] = this.prioritizer.getNotLocalCpy();
+        const localCpyExists: boolean = this.prioritizer.getLocalCpyURI().length > 0;
 
-        if (!message.length) {
-            missingCopybooks.forEach(copybook => this.queue.push(copybook, profile));
-        } else if (missingCopybooks.length > 0) {
-            this.resolver.fixMissingDownloads(message, missingCopybooks, profile, {
-                hasPaths: (await this.pathGenerator.listDatasets()).length > 0,
-                hasProfiles: Object.keys(await this.profileService.listProfiles()).length > 1,
-            });
+        // message to be displayed should be adjusted according to the requirments
+        if (localCpyExists) {
+        // we can use the logic from continueDownloadFromMF in order to ask user if would like to use Zowe or no - TBD
+                this.notifyUserIfCopyNotLocal("Some copybooks are not present locally", notLocalCpy.length > 0);
```

Comment:
Why we have to pass a boolean in the method? If at some point we are sure that we have to throw an error message I don't see any reason to apply a boolean check

Reply:
I will answer here for both, this is an enabler and my idea behind is not to throw a message to the user unless necessary. I apply a check in order to make sure that I do have copybooks not present in the file, this method needs to be adapted to the Zowe scenario, the message is redundant for now and has to be changed accordingly.